src/core/Pagination/index.tsx

```diff
@@ -0,0 +1,166 @@
+  StyledPaginationDropdownMenu,
+} from "./style";
+import { usePagination } from "./usePagination";
+
+export interface PaginationInternalProps {
+  onPageChange: (page: number | string) => void;
+  onNextPage: () => void;
+  onPreviousPage: () => void;
+  totalCount: number;
+  siblingCount?: number;
+  currentPage: number;
+  pageSize: number;
+  truncateDropdown?: boolean;
+}
+
+export type PaginationProps = PaginationExtraProps & PaginationInternalProps;
+
+interface PageListDropdownProps {
+  pageList: number[];
+  onPageChange: (page: number | string) => void;
+  sdsStyle: "round" | "square";
+}
+
+const PageListDropdown = (props: PageListDropdownProps) => {
+  const { pageList, onPageChange, sdsStyle } = props;
+
+  const [anchorEl, setAnchorEl] = React.useState<Element | null>(null);
+
+  const handleClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <Page sdsStyle={sdsStyle}>
+      <StyledPaginationButtonIcon onClick={(e) => handleClick(e)}>
+        <Icon sdsIcon="dotsHorizontal" sdsSize="xs" sdsType="iconButton" />
+      </StyledPaginationButtonIcon>
+      <StyledPaginationDropdownMenu
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        {pageList.map((page: number) => {
+          return (
+            <MenuItem
+              onClick={() => {
+                onPageChange(page);
+                handleClose();
+              }}
+              key={page}
+            >
+              {page}
+            </MenuItem>
+          );
+        })}
+      </StyledPaginationDropdownMenu>
+    </Page>
+  );
+};
+
+const Pagination = forwardRef<HTMLUListElement, PaginationProps>(
+  (props: PaginationProps, ref): JSX.Element | null => {
+    const {
+      sdsStyle = "round",
+      onPageChange,
+      onNextPage,
+      onPreviousPage,
+      totalCount,
+      siblingCount = 1,
+      currentPage,
+      pageSize,
+      truncateDropdown = true,
+    } = props;
+
+    if (pageSize < 1) {
+      throw new Error("PageSize can not be smaller than 1!");
+    }
+
+    const paginationRange =
+      usePagination({
+        currentPage,
+        pageSize,
+        siblingCount,
+        totalCount,
+      }) || [];
+
+    if (currentPage === 0 || paginationRange.length < 2) {
+      return null;
+    }
+
+    const lastPage = paginationRange[paginationRange.length - 1];
+
+    return (
+      <StyledPagination ref={ref} {...props}>
+        <StyledPaginationChevronButton
+          key="prevPage"
+          onClick={() => currentPage > 1 && onPreviousPage()}
+          disabled={currentPage === 1}
+          sdsSize="small"
+        >
+          <Icon sdsIcon="chevronLeft" sdsType="iconButton" sdsSize="s" />
+        </StyledPaginationChevronButton>
+
+        {paginationRange.map((pageNumber) => {
```

Comment:
does this mean the developer will have to tell the paginator how to truncate? For example, I would pass 
```
[1, 2, 3
    [4, 5, 6, 7],
8, 9]
```

to get a truncation for pages 4-7?